apps/mark-scan/backend/src/app.ts

```diff
@@ -321,11 +322,20 @@ export function buildApi(
       store.setBallotsPrintedCount(0);
     },
 
-    setPrecinctSelection(input: {
+    async setPrecinctSelection(input: {
       precinctSelection: PrecinctSelection;
-    }): void {
+    }): Promise<void> {
+      const electionDefinition = store.getElectionDefinition();
+      assert(electionDefinition);
       store.setPrecinctSelection(input.precinctSelection);
       store.setBallotsPrintedCount(0);
+      await logger.logAsCurrentUser(LogEventId.PrecinctConfigurationChanged, {
+        disposition: 'success',
+        message: `User set the precinct for the machine to ${getPrecinctSelectionName(
+          electionDefinition.election.precincts,
+          input.precinctSelection
+        )}`,
+      });
```

Comment:
agnostic as to doing it in this PR, but little logging bits like this could be broken out into logging helpers in `utils` to make it a little less verbose in app code.

Reply:
I think we should add an API middleware that logs every request, its inputs, and whether it succeeded or failed. That would cut out a lot of this verbosity.